docs/timezone.md

```diff
@@ -27,6 +27,12 @@ The object must have `getOffsetNanosecondsFor()`, `getPossibleInstantsFor()`, an
 It must not have a `timeZone` property, so that it can be distinguished in `Temporal.TimeZone.from()` from other Temporal objects that have a time zone.
 It is possible to pass such an object into any Temporal API that would normally take a built-in `Temporal.TimeZone`.
 
+Any implementation of the protocol with those three methods will return the correct output from any Temporal property or method.
+However, most other code will assume that custom time zones act like built-in `Temporal.TimeZone` objects.
+If you want to interoperate with libraries or other code that you didn't write, then you should implement all the other `Temporal.TimeZone` methods and properties described below as well: `id`, `getOffsetStringFor()`, `getDateTimeFor()`, `getZonedDateTimeFor()`, `getInstantFor()`, `getNextTransition()`, `getPreviousTransition()`, and `toJSON()`.
+
+Creating a class that inherits from the built-in `Temporal.TimeZone` and overriding the methods you need to override, instead of using a plain object that implements the time zone protocol, is by far the easiest (and recommended) way to achieve this.
```

Comment:
We could probably simplify (and maybe shrink) the entire custom timezones section by re-ordering the content so that you explain which approach is easier at the top, instead of at the bottom.  Here's some suggested content to show what I mean.

## Custom Time Zones

For specialized applications where you need to do calculations in a time zone that is not built into Temporal, you can implement a custom time zone.
There are two ways to do this.

The easiest way is to create a class inheriting from `Temporal.TimeZone`, call `super()` in the constructor with a time zone identifier string, and implement  `getOffsetNanosecondsFor()` and `getPossibleInstantsFor()`.
You can optionally also implement `getNextTransition()` and `getPreviousTransition()`, or leave the default implementation which always returns `null` which means that no next or previous transitions are known.
You don't need to implement any other methods because other properties and methods will be automatically implemented via the base class calling the methods that you override.

The harder way to create a custom time zone is to create a plain object without subclassing.
The object must at least implement `getOffsetNanosecondsFor()`, `getPossibleInstantsFor()`, and `toString()`.
An object with those three methods will return correct output from any Temporal property or method.
However, most other code will assume that custom time zones act like built-in `Temporal.TimeZone` objects.
To interoperate with libraries or other code that you didn't write, then you should implement all other `Temporal.TimeZone` members as well: `id`, `getOffsetStringFor()`, `getDateTimeFor()`, `getZonedDateTimeFor()`, `getInstantFor()`, `getNextTransition()`, `getPreviousTransition()`, and `toJSON()`.
Your object must not have a `timeZone` property, so that `Temporal.TimeZone.from()` can distinguish your custom time zone from named-parameter input objects like `{ timeZone: 'Asia/Tokyo' }`.

The identifier of a custom time zone must consist of one or more components separated by slashes (`/`), as described in the [tzdata documentation](https://htmlpreview.github.io/?https://github.com/eggert/tz/blob/master/theory.html#naming).
Each component must consist of between one and 14 characters.
Valid characters are ASCII letters, `.`, `-`, and `_`.
`-` may not appear as the first character of a component, and a component may not be a single dot `.` or two dots `..`.

Reply:
I've adopted a slightly edited version of this.